File: r/bob/bob.R
bob <- function(input) {
  trimmed.input <- trimws(input);

  if (input == toupper(trimmed.input) && grepl("[A-Za-z]", trimmed.input)){
    return('Whoa, chill out!')
  }
  
  if (endsWith(trimmed.input, "?")) {
    return('Sure.')
  }
  
  if (trimmed.input == '') {
    return("Fine. Be that way!")  # ^\\s*$" asks for 0 or more (*) spaces (\\s) between beginning (^) and end ($) of string.
  }
  
  return('Whatever.')

}

File: python/bob/bob.py
def hey(conversation):
	if (conversation.isupper()):
		return 'Whoa, chill out!'
	elif (conversation.strip() == ''):
		return 'Fine. Be that way!'
	elif (conversation.strip().endswith('?')):
		return 'Sure.'
	else:
		return 'Whatever.'

print(hey(''))